network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/CalculatedBus.java

```diff
@@ -131,13 +137,94 @@ public double getV() {
+    }
+
+    private void setAngleInCalculatedBus(CalculatedBusAttributes calculatedBusAttributes, double value) {
+        calculatedBusAttributes.setAngle(value);
+    }
+
+    private void setAngleInConfiguredBus(ConfiguredBusAttributes configuredBusAttributes, double value) {
+        configuredBusAttributes.setAngle(value);
+    }
+
+    private double getAngleInBus(Bus bus) {
+        return bus.getAngle();
+    }
+
     @Override
     public Bus setV(double v) {
         getAttributes().setV(v);
         index.updateVoltageLevelResource(voltageLevelResource, AttributeFilter.SV);
+
+        if (getVoltageLevel().getTopologyKind() == TopologyKind.BUS_BREAKER) {
+            // update V in configured buses
+            updateConfiguredBuses(v, getAttributes(), VOLTAGE, this::getVInBus, this::setVInConfiguredBus);
+        } else {
+            if (isBusView) {
+                // update V for buses in BusBreakerView
+                updateBusesAttributes(v, voltageLevelResource.getAttributes().getCalculatedBusesForBusBreakerView(), getAttributes(), this::setVInCalculatedBus);
+            } else {
+                // update V for buses in BusView
+                updateBusesAttributes(v, voltageLevelResource.getAttributes().getCalculatedBusesForBusView(), getAttributes(), this::setVInCalculatedBus);
+            }
+        }
         return this;
     }
 
+    private void updateBusesAttributes(double value,
+                                       List<CalculatedBusAttributes> calculatedBusAttributesList,
+                                       CalculatedBusAttributes sourceBusAttributes,
+                                       ObjDoubleConsumer<CalculatedBusAttributes> setValue) {
+        if (!CollectionUtils.isEmpty(calculatedBusAttributesList)) {
+            calculatedBusAttributesList.forEach(busToUpdate -> busToUpdate.getVertices().forEach(vertex1 ->
+                sourceBusAttributes.getVertices().stream().filter(v -> v.getId().equals(vertex1.getId())).findFirst().ifPresent(vertex2 -> {
+                    setValue.accept(busToUpdate, value);
+                    index.updateVoltageLevelResource(voltageLevelResource, AttributeFilter.SV);
+                })
+            ));
+        }
+    }
+
+    private void updateConfiguredBuses(double newValue,
+                                       CalculatedBusAttributes calculatedBusAttributes,
+                                       String attributeName,
+                                       ToDoubleFunction<Bus> getValue,
+                                       ObjDoubleConsumer<ConfiguredBusAttributes> setValue) {
+        List<String> busesIds = calculatedBusAttributes.getVertices().stream()
+            .map(Vertex::getBus)
+            .toList();
+
+        List<Bus> buses = getVoltageLevel().getBusBreakerView().getBusStream()
+            .filter(bus -> busesIds.contains(bus.getId()) && !Objects.equals(getValue.applyAsDouble(bus), newValue))
+            .toList();
```

Comment:
```suggestion
        List<Bus> buses = ((VoltageLevelImpl) getVoltageLevel()).getResource().getAttributes()
                .getBusToCalculatedBusForBusView().entrySet().stream()
                .filter(entry -> getCalculatedBusNum() == entry.getValue())
                .map(Map.Entry::getKey)
                .map(getVoltageLevel().getBusBreakerView()::getBus)
                .toList();

```
something like that ?